torch/nested/codegen/extension.py

```diff
@@ -0,0 +1,216 @@
+    return [
+        'add',
+        'mul',
+        'sub',
+        'div',
+        'pow',
+        'atan2',
+        'remainder',
+    ]
+
+
+def get_comparison_functions():
+    return [
+        'eq',
+        'ge',
+        'gt',
+        'le',
+        'ne',
+        'lt'
+    ]
+
+
+# Adds dispatch based on isinstance to torch function
+# tfunc by wrapping the current torch.tfunc function
+# and overwriting it with a function that dispatches
+# to func(tfunc, reference_to_tfunc, *args, **kwargs)
+# if the first argument is of instance cls.
+# NOTE: This function is a hack. This dispatch mechanism
+# is slow and should not be used in release code, but for
+# development only.
+def set_function(module, cls, tfunc, func):
+    def _gen_func(tfunc):
+        orig_tfunc = getattr(torch, tfunc)
+
+        @wraps(orig_tfunc)
+        def _func(*args, **kwargs):
+            if isinstance(args[0], cls):
+                return func(*((tfunc, orig_tfunc,) + args), **kwargs)
+            else:
+                return orig_tfunc(*args, **kwargs)
+        return _func
+    setattr(module, tfunc, _gen_func(tfunc))
+
+
+def _check_meaningful_overwrite(cls, method_name):
+    class DefaultClass(object):
+        pass
+
+    if getattr(cls, method_name, False) and not getattr(DefaultClass, method_name, False):
+        raise Exception("WARNING: " + method_name + " already exists "
+                        "and not part of default class")
+
+
+def set_unary_method(cls, tfunc, method_name, inplace):
+    def _gen_func(method_name):
+        def _func(self):
+            assert isinstance(self, cls)
+            if inplace:
+                return getattr(torch, tfunc)(self, out=self)
+            else:
+                return getattr(torch, tfunc)(self)
+        return _func
+    _check_meaningful_overwrite(cls, method_name)
+    setattr(cls, method_name, _gen_func(method_name))
+
+
+def set_binary_method(cls, tfunc, method_name, inplace):
+    def _gen_func(tfunc):
+        def _func(self, other):
+            if inplace:
+                return getattr(torch, tfunc)(self, other, out=self)
+            else:
+                return getattr(torch, tfunc)(self, other)
+        return _func
+    _check_meaningful_overwrite(cls, method_name)
+    setattr(cls, method_name, _gen_func(tfunc))
+
+
+def get_unary_method_signatures():
+    signatures = []
+    for method_name in get_unary_functions():
+        signatures.append(Signature(method_name, method_name, False))
+        signatures.append(Signature(method_name, method_name + "_", True))
+    return signatures
+
+
+def get_binary_method_signatures():
+    signatures = []
+    for method_name in get_binary_functions():
+        signatures.append(Signature(method_name, method_name, False))
+        signatures.append(Signature(method_name, method_name + "_", True))
+
+    for method_name in ['add', 'mul', 'sub', 'div']:
+        signatures.append(Signature(method_name, "__" + method_name + "__", False))
```

Comment:
There are also `r` methods (`radd`, `rsub`), etc but I"m not sure if you want to handle them now because those deal with NestedTensor <-> other type interop.

Reply:
Good point. Not for now, because it's not covered by the RFC. Tensor <-> NestedTensor interoperability however will need to be covered soon